opencog/cogserver/modules/events/AtomSpacePublisherModule.h

```diff
@@ -81,91 +81,91 @@ class AtomSpacePublisherModule;
-           false, false)
-
-    public:
-        AtomSpacePublisherModule(CogServer&);
-        virtual ~AtomSpacePublisherModule();
-        virtual void run();
-
-        static const char *id(void);
-        virtual void init(void);
-
-        void atomAddSignal(Handle h);
-        void atomRemoveSignal(AtomPtr atom);
-        void AVChangedSignal(const Handle& h,
-                             const AttentionValuePtr& av_old,
-                             const AttentionValuePtr& av_new);
-        void TVChangedSignal(const Handle& h,
-                             const TruthValuePtr& tv_old,
-                             const TruthValuePtr& tv_new);
-        void addAFSignal(const Handle& h,
-                         const AttentionValuePtr& av_old,
-                         const AttentionValuePtr& av_new);
-        void removeAFSignal(const Handle& h,
-                         const AttentionValuePtr& av_old,
-                         const AttentionValuePtr& av_new);
+	private:
+		AtomSpace* as;
+		AttentionBank* _attention_bank;
+
+		AtomPtrSignal* _remove_atom_signal;
+		int _remove_atom_connection;
+
+		AtomSignal* _add_atom_signal;
+		int _add_atom_connection;
+
+		TVCHSigl* _tvchange_signal;
+		int _tvchange_connection;
+
+		AVCHSigl* _avchange_signal;
+		int _avchange_connection;
+
+		AVCHSigl* _add_af_signal;
+		int _add_af_connection;
+
+		AVCHSigl* _remove_af_signal;
+		int _remove_af_connection;
+
+		void enableSignals();
+		void disableSignals();
+
+		// TBB
+		tbb::concurrent_bounded_queue<message_t> queue;
+
+		// ZeroMQ
+		zmq::context_t * context;
+		void InitZeroMQ();
+		void proxy();
+
+		void sendMessage(std::string messageType, std::string payload);
+		std::string atomMessage(Json::Value jsonAtom);
+		std::string avMessage(Json::Value jsonAtom, Json::Value jsonAVOld, Json::Value jsonAVNew);
+		std::string tvMessage(Json::Value jsonAtom, Json::Value jsonTVOld, Json::Value jsonTVNew);
+		Json::Value atomToJSON(Handle h);
+		Json::Value tvToJSON(TruthValuePtr tv);
+		Json::Value avToJSON(AttentionValuePtr av);
+		// TODO: add protoatom to JSON functionality
+
+		DECLARE_CMD_REQUEST(AtomSpacePublisherModule, "publisher-enable-signals",
+			 do_publisherEnableSignals,
+			 "Enable AtomSpace event publishing",
+			 "Usage: publisher-enable-signals",
+			 false, false)
+
+		DECLARE_CMD_REQUEST(AtomSpacePublisherModule, "publisher-disable-signals",
+			 do_publisherDisableSignals,
+			 "Disable AtomSpace event publishing",
+			 "Usage: publisher-disable-signals",
+			 false, false)
+
+	public:
+		AtomSpacePublisherModule(CogServer&);
+		virtual ~AtomSpacePublisherModule();
+		virtual void run();
+
+		static const char *id(void);
+		virtual void init(void);
+
+		void atomAddSignal(Handle h);
+		void atomRemoveSignal(AtomPtr atom);
+		void AVChangedSignal(const Handle& h,
+							 const AttentionValuePtr& av_old,
+							 const AttentionValuePtr& av_new);
+		void TVChangedSignal(const Handle& h,
```

Comment:
The indentation here looks incorrect. I think you want to use tabs, three, I guess, to line up with `void AVChangedSignal(const Handle& h,` and then use spaces the rest of the way, so that `const AttentionValuePtr& av_old,` lines up correctly, no matter what size the tabs are.

Reply:
that's what sucks about using tabs over spaces... changing tabstop results in inconsistent view. I was using tabstop=4 on my vim and it looked well aligned until the github viewer showed it with 8 spaces then it's ugly.

Reply:
@Dagiopia, please see http://vim.wikia.com/wiki/Indent_with_tabs,_align_with_spaces

Reply:
My recommendation is that you set tabstap to 3 or 5, that way if you stumble on a space indented file (that still concerns a third of the code), you'll immediately catch it.

Reply:
pushed the changes on #3311 

thanks @linas and @ngeiswei